app/services/requirements/datetime_checker.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+module Requirements
+  class DatetimeChecker
+    attr_reader :day, :month, :year, :hour, :minutes, :meridien
+
+    MAXIMUM_DAY = 31
+    MINIMUM_DAY = 1
+    MAXIMUM_MONTH = 12
+    MINIMUM_MONTH = 1
+    MINIMUM_YEAR = Time.zone.now.year
+    MAXIMUM_HOUR = 12
+    MINIMUM_HOUR = 1
+    MAXIMUM_MINUTES = 59
+    MERIDIEN_VALUES = %w(am pm).freeze
+    FUTURE_TIME_PERIOD = { months: 14 }.freeze
+
+    def initialize(params)
```

Comment:
Do we have a requirement to give specific errors for each field? Especially as the fields are only on a couple of lines in the final design, I think it's probably enough to use `Date.parse` and `Time.parse` and just tell them they've entered an invalid date/time.

Reply:
I just felt like it would be helpful for users to know exactly what is wrong with their input so they can correct it.

Reply:
I think we'll struggle to display that many errors. If a user left everything blank they'd have at least 3 errors given which could all be summarised with "Enter a valid date". Particularly as we have very little space.

Might be worth checking with design but I think we could probably just have "Enter a valid date" for blank or any odd input, "Enter a date and time in the future" for that scenario and the time period one. 

The aspect that I think could be useful with this level of error messages is informing a user they entered the wrong format (e.g. text when we expected a number) - but then I don't really feel validation errors is the way to inform them how they should enter text and if that's ambiguous it should be clearer in the UI. Sidenote - I wonder if we want to use input type=number, datalist or both which may make that clearer.

Reply:
I've reduced this down to just checking the validity of the date and time, and that the datetime provided isn't in the past or too far in the future.